packages/firefuel/test/src/firefuel_collection_test.dart

```diff
@@ -349,6 +398,60 @@ void main() {
       expect(readResult, updatedDoc);
     });
   });
+
+  group('#where', () {
+    const String expectedName = 'expectedName',
+        unexpectedName1 = 'unexpectedName1',
+        unexpectedName2 = 'unexpectedName2';
+    final expectedUser = TestUser(expectedName);
+
+    setUp(() async {
+      await testCollection.create(TestUser(unexpectedName1));
+      await testCollection.create(expectedUser);
+      await testCollection.create(TestUser(unexpectedName2));
+    });
+
+    test('should return a subset of the existing list', () async {
+      final filteredList = await testCollection.where(
+        [Clause(TestUser.fieldName, isEqualTo: expectedName)],
+      );
+
+      expect(filteredList, [expectedUser]);
+    });
+
+    test('should return a subset based on multiple clauses', () async {
+      final filteredList = await testCollection.where(
+        [
+          Clause(TestUser.fieldName, isNotEqualTo: unexpectedName1),
+          Clause(TestUser.fieldName, isNotEqualTo: unexpectedName2),
+        ],
+      );
+
+      expect(filteredList, [expectedUser]);
+    });
+
+    test('should return an empty list when more than one option is chosen',
+        () async {
+      final filteredList = await testCollection.where(
+        [
+          Clause(
+            TestUser.fieldName,
+            isNotEqualTo: unexpectedName1,
+            isEqualTo: expectedUser,
+          ),
+        ],
+      );
+
+      expect(filteredList, []);
+    });
```

Comment:
Although I think that this may be the default firebase experience, I think that it may be confusing to get an empty list rather than an error.

If I get an empty list from my clause, I would think that I don't have anything in my collection that matches the clause, which could be false. Meaning that I know that I have `expectedUser` in my collection and I know for a fact that its not equal to `unexpectedName1`, but I am getting an empty list in return.

If we were to throw an exception instead, it could give more context to the user as to what to do next to get a list of values.
**Or**
Instead of arguments for each input, we were to create named constructors for each input that expected a value?

Another option, is that we allow multiple inputs, such as this, and we create each individual input as its own unique clause if it has any value assigned.

Reply:
Let's go with the exception route so it's explicit that each clause stands alone